soccer/planning/RRTPlanner.cpp

```diff
@@ -180,47 +184,57 @@ Planning::InterpolatedPath RRTPlanner::makePath()
 	return newPath;
 }
 
-void RRTPlanner::optimize(Planning::InterpolatedPath &path, const Geometry2d::CompositeShape *obstacles, const MotionConstraints &motionConstraints, Geometry2d::Point vi)
+Planning::InterpolatedPath* RRTPlanner::optimize(Planning::InterpolatedPath &path, const Geometry2d::CompositeShape *obstacles, const MotionConstraints &motionConstraints, Geometry2d::Point vi)
```

Comment:
True, but the fact that it's deleting the Path that's passed in by reference is pretty bad.  For all this function knows, the Path could be stack allocated and deleting it would result in immediate program termination :bomb:.  Can we have it modify in place and return a bool false on failure?  We'll nuke this class later, but for this PR, let's at least make the interface make sense.  The goal is to always keep the master branch decent even if we plan to improve it later :)